src/subdomains/supporting/pricing/domain/entities/price-rule.entity.ts

```diff
@@ -0,0 +1,118 @@
+import { Asset } from 'src/shared/models/asset/asset.entity';
+import { IEntity } from 'src/shared/models/entity';
+import { Fiat } from 'src/shared/models/fiat/fiat.entity';
+import { Column, Entity, ManyToOne, OneToMany } from 'typeorm';
+import { Price } from './price';
+
+export enum PriceSource {
+  KRAKEN = 'Kraken',
+  BINANCE = 'Binance',
+  COIN_GECKO = 'CoinGecko',
+}
+
+export interface Rule {
+  source: PriceSource;
+  asset: string;
+  reference: string;
+  limit?: number;
+}
+
+@Entity()
+export class PriceRule extends IEntity {
+  @OneToMany(() => Asset, (a) => a.priceRule)
+  assets: Asset[];
+
+  @OneToMany(() => Fiat, (f) => f.priceRule)
+  fiats: Fiat[];
+
+  @ManyToOne(() => Asset, { eager: true, nullable: false })
+  reference: Asset;
+
+  @Column()
+  priceSource: PriceSource;
+
+  @Column()
+  priceAsset: string;
+
+  @Column()
+  priceReference: string;
+
+  // check 1
+  @Column({ nullable: true })
+  check1Source: PriceSource;
+
+  @Column({ nullable: true })
+  check1Asset: string;
+
+  @Column({ nullable: true })
+  check1Reference: string;
+
+  @Column({ type: 'float', nullable: true })
+  check1Limit: number;
+
+  // check 2
+  @Column({ nullable: true })
+  check2Source: PriceSource;
+
+  @Column({ nullable: true })
+  check2Asset: string;
+
+  @Column({ nullable: true })
+  check2Reference: string;
+
+  @Column({ type: 'float', nullable: true })
+  check2Limit: number;
+
+  // price
+  @Column({ type: 'float', nullable: true })
+  currentPrice: number;
+
+  @Column({ type: 'integer' })
+  priceValiditySeconds: number;
+
+  @Column({ type: 'datetime2', nullable: true })
+  priceTimestamp: Date;
+
+  // getters
+  get isPriceValid(): boolean {
+    return (
+      this.currentPrice != null &&
+      this.priceTimestamp != null &&
+      (Date.now() - this.priceTimestamp.getTime()) / 1000 <= this.priceValiditySeconds
```

Comment:
`Util.secondsDiff(this.priceTimestamp, Date.now()) <= this.priceValiditySeconds`?